blazor/dropdown-list/data-binding.md

```diff
@@ -7,68 +7,377 @@ control: DropDown List
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/data-binding/offline-mode.razor %}
+
+{% endhighlight %}
+
+![Blazor DropdownList with Offline mode](./images/data-binding/blazor_dropdown_offline-mode.png)
+
+### Entity framework
 
-Index value binding can be achieved by using `bind-Index` attribute and it supports int and int nullable types. By using this attribute you can bind the values respective to its index.
+You need to follow the below steps to consume data from the [Entity Framework](https://blazor.syncfusion.com/documentation/common/data-binding/bind-entity-framework) in the DropDownList component.
 
-```cshtml
-@using Syncfusion.Blazor.DropDowns
+#### Create DBContext class
 
-<SfAutoComplete TValue="string" Placeholder="e.g. Australia" TItem="Country" @bind-Index="@ddlIndex" DataSource="@Countries">
-    <AutoCompleteFieldSettings Value="Name"></AutoCompleteFieldSettings>
-</SfAutoComplete>
+The first step is to create a DBContext class called `OrderContext` to connect to a Microsoft SQL Server database.
 
-@code {
+```csharp
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EFDropDown.Shared.Models;
 
-    private int? ddlIndex { get; set; } = 1;
+namespace EFDropDown.Shared.DataAccess
+{
+    public class OrderContext : DbContext
+    {
+        public virtual DbSet<Shared.Models.Order> Orders { get; set; }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+                optionsBuilder.UseSqlServer(@"Data Source=(LocalDB)\MSSQLLocalDB;AttachDbFilename=D:\Blazor\DropDownList\EFDropDown\Shared\App_Data\NORTHWND.MDF;Integrated Security=True;Connect Timeout=30");
+            }
+        }
+    }
+}
+```
+
+#### Create data access layer to perform data operation
+
+Now, you need to create a class named `OrderDataAccessLayer`, which act as data access layer for retrieving the records from the database table.
+
+```csharp
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EFDropDown.Shared.Models;
 
-    public class Country
+namespace EFDropDown.Shared.DataAccess
+{
+    public class OrderDataAccessLayer
     {
-        public string Name { get; set; }
+        OrderContext db = new OrderContext();
 
-        public string Code { get; set; }
+        //To Get all Orders details
+        public DbSet<Order> GetAllOrders()
+        {
+            try
+            {
+                return db.Orders;
+            }
+            catch
+            {
+                throw;
+            }
+        }
     }
+}
+```
+
+#### Creating web API controller
 
-    List<Country> Countries = new List<Country>
+ A Web API Controller has to be created which allows DropDownList directly to consume data from the Entity framework.
```

Comment:
A Web API Controller has to be created, which allows the DropDownList to directly consume data from the Entity framework.